src/lib/search-and-replace.ts

```diff
@@ -170,34 +162,25 @@ export const searchAndReplace = async (
 	try {
 		replacedStream = await replace( readStream, replacements, binary );
 	} catch ( replaceError ) {
-		exit.withError( replaceError );
+		// replace() throws strings... OMG
+		exit.withError( replaceError as string | Error );
 	}
 
-	const result = await new Promise( ( resolve, reject ) => {
-		replacedStream
-			.pipe( writeStream )
-			.on( 'finish', () => {
-				resolve( {
-					inputFileName: fileName,
-					outputFileName,
-					usingStdOut,
-				} );
-			} )
-			.on( 'error', () => {
-				console.log(
-					chalk.red(
-						"Oh no! We couldn't write to the output file.  Please check your available disk space and file/folder permissions."
-					)
-				);
-
-				reject();
-			} );
-	} );
+	try {
+		await pipeline( replacedStream, writeStream );
+	} catch ( error ) {
+		console.log( red( "Oh no! We couldn't write to the output file.  Please check your available disk space and file/folder permissions." ) );
+		throw error;
+	}
```

Comment:
I love the `pipeline()` function :-) It was introduced in Node 15; we require at least Node 16.